presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotPageSourceProvider.java

```diff
@@ -127,4 +131,20 @@ public ConnectorPageSource createPageSource(
                 throw new UnsupportedOperationException("Unknown Pinot split type: " + pinotSplit.getSplitType());
         }
     }
+
+    @VisibleForTesting
+    static GrpcQueryClient.Config extractGrpcQueryClientConfig(PinotConfig config) {
+        Map<String, Object> target = new HashMap<>();
+        target.put(CONFIG_USE_PLAIN_TEXT, !config.isUseTlsForGrpc());
+        target.put(CONFIG_MAX_INBOUND_MESSAGE_BYTES_SIZE, config.getStreamingServerGrpcMaxInboundMessageBytes());
+        if (config.isUseTlsForGrpc()) {
+            target.put("tls.keystore.path", config.getGrpcTlsKeyStorePath());
+            target.put("tls.keystore.password", config.getGrpcTlsKeyStorePassword());
+            target.put("tls.keystore.type", config.getGrpcTlsKeyStoreType());
+            target.put("tls.truststore.path", config.getGrpcTlsTrustStorePath());
```

Comment:
If our Pinot Proxy is using Public CA signed certificate, this one will not be needed. Otherwise we need to manually add the Pinot Proxy hostname self-sign CA into this.